tests/test_app/test_composable.py

```diff
@@ -634,5 +633,34 @@ def main(self, val: int) -> int:
     __app_registry.pop(get_object_provenance(app_decorated_first1))
 
 
+# have to define this at module level for pickling to work
+@define_app
+def func2app(arg1: int, exponent: int) -> float:
+    return arg1 ** exponent
+
+
+def test_decorate_app_function():
+    """works on functions now"""
+    import inspect
+
+    sqd = func2app(exponent=2)
+    assert sqd(3) == 9
+    assert inspect.isclass(func2app)
+    p = get_object_provenance(func2app)
+    __app_registry.pop(p, None)
+
+
+def test_roundtrip_decorated_function():
+    """decorated function can be pickled/unpickled"""
+    import pickle
+
+    sqd = func2app(exponent=2)
+    u = pickle.loads(pickle.dumps(sqd))
+    assert u(4) == 16
+
+    p = get_object_provenance(func2app)
+    __app_registry.pop(p, None)
+
+
```

Comment:
After you merged this, you can make a new issue to add more tests. I will do it.